### DISCUSSION TITLE
"<ItemsControl Items=*>" vs "<ItemsControl ItemsSource=*>"

### DISCUSSION THREAD

**Discussion by HendrikHoemberg** [AvaloniaUI/Avalonia#13316] [Q&A] [answered]
Hi! 

I'm very new to coding and i've encountered a problem in the Avalonia "ToDo-List tutorial" that made me pull my hair out for 1 hour.

In the step "Wiring up the Views" at the point were we update the TodoListView, the code that should be working says 
`<ItemsControl Items="{Binding Items}"> ... </ItemsControl>`

But when i've tried to run it, i got the 
"Error Unable to find suitable setter or adder for property Items of type Avalonia.Controls:Avalonia.Controls.ItemsControl for argument Avalonia.Markup:Avalonia.Data.Binding..."

I've tried many things when i finally found the solution by giving me alternative suggestions and by trying out "ItemsSource" instead of "Items", it finally worked. So i thought that maybe the Items property got removed but when i looked up the documentation, it's still stated as a property. 

Could anyone explain to me why Items doesn't work and what's the difference between Items and ItemsSource

thx in advance!

P.S.: Very small issue but maybe the tutorial could get updated to explicitly tell, that you should uncheck Enabled Compiled Bindings when creating the project. VIsual Studio enabled Compiled Bindings by default and i had to manually disable it to make the MainWindow be able to fetch the Binding Content from the list.

**Comment by maxkatz6** [comment]
It seems this part of the tutorial wasn't updated for 11.0.

> <ItemsControl Items="{Binding Items}"> ... </ItemsControl>

As you already found out, it should be:
```
<ItemsControl ItemsSource="{Binding Items}"> ... </ItemsControl>
```

The difference in 11.0 - `ItemsControl.ItemsSource` is for binding and setting item sources.  Items is a readonly collection property that is pre-populated, and ItemsSource is the read/write version that has a default value of null. When you add any items to the control from the XAML (without bindings), it will be added to the Items collection only.

> you should uncheck Enabled Compiled Bindings when creating the project

Well, you can, but it's not really a recommended approach.

**Reply by maxkatz6** [reply, marked as answer]
Update: it looks like everything is good, but you used wrong documentation version.

`Wiring up the Views` you mentioned is from 0.10 version of documentation: https://docs.avaloniaui.net/docs/tutorials/todo-list-app/wiring-up-the-views

In 11.0 version this tutorial was restructured: https://docs.avaloniaui.net/docs/next/tutorials/todo-list-app/wiring-up-the-views, and it uses ItemsSource.

Possibly compiled bindings issue is also related to the same problem.

**Reply by HendrikHoemberg** [reply]
Damn, now i feel dumb... i was using the old 0.10 tutorial the whole time.

Well, thx for the clarification, i'll now take the walk of shame and use the newest version xD

Thank you very much!

**Reply by thevortexcloud** [reply]
If it makes you feel better its probably not completely your fault. Google still points to the .10 documentation when you search for pretty much anything. Someone probably needs to fiddle with some SEO stuff. I am guessing it's because the 11.0 documentation has "next" in the URL which sounds very much like a beta thing.
